recipes/libmikmod/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_oss
+            del self.options.with_pulse
+        # Apple
+        if tools.is_apple_os(self.settings.os):
+            del self.options.with_coreaudio
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.settings.os == "Linux":
+            if self.options.with_alsa:
+                self.requires("libalsa/1.1.9")
+            if self.options.with_pulse:
+                self.requires("pulseaudio/13.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self, set_cmake_flags=True)
+        cmake.definitions["ENABLE_STATIC"] = not self.options.shared
+        cmake.definitions["ENABLE_DOC"] = False
+        cmake.definitions["ENABLE_DSOUND"] = self.options.get_safe("with_dsound", False)
+        cmake.definitions["ENABLE_MMSOUND"] = self.options.get_safe("with_mmsound", False)
+        cmake.definitions["ENABLE_ALSA"] = self.options.get_safe("with_alsa", False)
+        cmake.definitions["ENABLE_OSS"] = self.options.get_safe("with_oss", False)
+        cmake.definitions["ENABLE_PULSE"] = self.options.get_safe("with_pulse", False)
+        cmake.definitions["ENABLE_COREAUDIO"] = self.options.get_safe("with_coreaudio", False)
+        cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        # 0001:
+        #   Patch CMakeLists.txt to run `conan_basic_setup`, to avoid building shared lib when
+        #   shared=False, and a fix to install .dlls correctly on Windows
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+         # Ensure missing dependencies yields errors
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "MESSAGE(WARNING",
+                              "MESSAGE(FATAL_ERROR")
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "drivers", "drv_alsa.c"),
+                              "alsa_pcm_close(pcm_h);",
+                              "if (pcm_h) alsa_pcm_close(pcm_h);")
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        os.remove(os.path.join(self.package_folder, "bin", "libmikmod-config"))
+        if not self.options.shared:
+            tools.rmdir(os.path.join(self.package_folder, "bin"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
Should there be a filename set in package info?

Reply:
libmikmod.pc, the name is ok by default

Reply:
We started to explicitly set the pkg_config name as there were thoughts about not generating pc files when the name was not set.

Reply:
@ericLemanissier 
See https://github.com/conan-io/conan/issues/7560

Reply:
there is no decision taken about this. why `.name` and not `.filename` ? maybe it will end up being a separate `.generate=True`

Reply:
Correct, but by adding we know in the future that we need a name for the pkg_config generator.
Not adding means that we need to check again whether a file is actually generated and what its name is.
Imho, it's better to be explicit and do things to have less work in the future.

Reply:
Thanks, it was harmles anyways.